drivers/hmis/app/graphql/mutations/ce/submit_ce_referral_step.rb

```diff
@@ -31,16 +31,20 @@ def resolve(referral_id:, step_id:, form_definition_id:, input:, **_rest) # add
         step = engine.active_steps.find(step_id)
         step.form_definition = form_definition
 
+        # TODO(#7395): permission
+
         validations = engine.validate_step(step, submitted_values: input)
         errors.push(*validations)
         return { errors: errors } if errors.any?
 
         engine.complete_step!(step, user: current_user, submitted_values: input)
       end
 
+      referral.reload
+
       {
-        step: step,
-        referral: referral.reload,
+        step: OpenStruct.new(step: step, referral: referral),
```

Comment:
Suggestion to be safe, reload the step as well to ensure we are returning the current state. Since the engine is operating on the state (And we make an assignment to it on line 32) .

```suggestion
        step: OpenStruct.new(step: step.reload, referral: referral.reload),
```